lib/resources/virtualization.rb

```diff
@@ -0,0 +1,228 @@
+require 'hashie/mash'
+
+module Inspec::Resources
+  class Virtualization < Inspec.resource(1)
+    name 'virtualization'
+    desc 'Use the virtualization InSpec audit resource to test the virtualization platform on which the system is running'
+    example "
+      describe virtualization[:system] do
+        it { should eq 'docker' }
+      end
+
+      describe virtualization[:role] do
+        it { should eq 'guest' }
+      end
+
+      describe virtualization[:systems][:docker] do
+        it { should eq 'guest' }
+      end
+    "
+
+    def initialize()
+      unless inspec.os.linux?
+        skip_resource 'The `virtualization` resource is not supported on your OS yet.'
+      end
+    end
+
+    def [](name)
```

Comment:
Since you have proper methods defined below around like 45/46, can we remove this method now?